text/0782-deprecate-ember-data-ajax.md

```diff
@@ -0,0 +1,43 @@
+---
+Stage: In Progress
+Start Date: 2022-01-29
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): ember-data
+RFC PR: TODO
+---
+
+# EmberData | Deprecate ajax requests
+
+## Summary
+
+Deprecates use of `ajax` methods in favor of `fetch` to request API data.
+
+## Motivation
+
+Currently, `ember-data` maintains two methods for users to fetch data from their backend - fetching with `jQuery.ajax` and `fetch`.  Confusion further exists in the adapters if users implement functionality to change out of the box behaviour. They can override [`ajax`](https://github.com/emberjs/data/blob/e076e0ae71ae6426ca53ad3c5501a0af7ceca883/packages/adapter/addon/rest.ts#L1091-L1126) methods but without a clean interface to operate with, what they really may be doing is overriding how fetch works. 
+
+With Ember 4.0 dropping jQuery, we should take a hard stance on dropping the use of `ajax` in the next major release. 
+
+## Detailed design
+
+The first step is putting in place a deprecation if an adapter uses `ajax` to make a request. Since `ember-data` switched the default behaviour to use fetch in 4.0, this deprecation will only apply to those who have overrided the default with `useFetch = true` in their adapters.
+
+Second, we will expose fetch related methods on the adapters that are equivalent in use and flexibility as the current `ajax` options.  Note the [`minimum-adapter-interface`](https://github.com/emberjs/data/blob/master/packages/store/addon/-private/ts-interfaces/minimum-adapter-interface.ts) does not assert the method by which a request is made. As a result, how we design and implement these methods should not have an effect on those users who have implemented their own adapter. 
+
+In 5.0, all `ajax` methods and supporting infrastructure will be removed.
+
+## How we teach this
+
+This is best taught through a deprecation guide. Users using `ajax` will need to install `ember-fetch` or rely on the default global `fetch` API provided by the browser.  This has some implications, for example, with how query params are serialized.  Users will have to be careful to ensure the behaviour of their API changes with any changes that result from switching to fetch.
+
+## Drawbacks
+
+Some user's technology stacks may be entirely dependent on `ajax` for requesting API data.  This isn't necessarily an easy switch and may require multiple improvements to various layers to use `fetch`. For those users, we can document how they can still implement their own adapter to use `ajax`.
+
```

Comment:
If you are depending on ajax, override fetch...apply to users dependent on this.

1. ajaxOptions and ajax prefilter
2. fetch method can use jquery.ajax in order to take advantage of Ajax setup and Ajax prefilter